Backend/nix/services.nix

```diff
@@ -0,0 +1,145 @@
+{ inputs, ... }:
+{
+  config = {
+    perSystem = { inputs', self', pkgs, lib, ... }: {
+      process-compose."services" = { config, ... }:
+        let
+          startScript = masterDb: replicaDb: port: pkgs.writeShellApplication {
+            name = "start-pgbasebackup";
+            runtimeInputs = with pkgs; [ config.services.postgres."${masterDb}".package coreutils ];
+            text = ''
+              masterDbPath=$(readlink -f data/${masterDb})
+              replicaDbPath=$(readlink -f data/${replicaDb})
+              pg_basebackup -h "$masterDbPath" -U repl_user --checkpoint=fast -D "$replicaDbPath"  -R --slot=some_name  -C --port=${port} 
+            '';
+          };
+          openStreetDataFile = inputs.osrm-pbf;
+          # NOTE: This *should* match the flake input.
+          openStreetDataFileName = "southern-zone-231218";
+          osrm-data =
+            pkgs.runCommandNoCC "osrm-data"
+              { buildInputs = [ pkgs.osrm-backend ]; }
+              ''
+                mkdir $out && cd $out
+                cp ${openStreetDataFile} ${openStreetDataFileName}.osm.pbf
+                osrm-extract -p ${pkgs.osrm-backend}/share/osrm/profiles/car.lua ${openStreetDataFileName}.osm.pbf
+                osrm-partition ${openStreetDataFileName}.osrm
+                osrm-customize ${openStreetDataFileName}.osrm
+              '';
+
+          osrm-server = pkgs.writeShellApplication {
+            name = "osrm-server";
+            runtimeInputs = [ pkgs.osrm-backend ];
+            text = ''
+              set -x
+              osrm-routed --algorithm mld \
+                ${osrm-data}/${openStreetDataFileName}.osrm
+            '';
+          };
```

Comment:
Note: there is already https://github.com/nammayatri/nammayatri/blob/main/Backend/nix/osrm.nix

Reply:
Ahh! I was about to replace that with `self’.packages.osrm-server`, totally forgot about it.